ibis/sql/postgres/client.py

```diff
@@ -65,19 +71,16 @@ def __init__(self, host=None, user=None, password=None, port=None,
         self.name = url.database
         self.database_name = 'public'
         self.con = sa.create_engine(url)
+        self.inspector = Inspector.from_engine(self.con)
         self.meta = sa.MetaData(bind=self.con)
 
     @property
     def current_database(self):
         return self.database_name
 
     def list_databases(self):
-        # http://dba.stackexchange.com/a/1304/58517
-        return [
-            row.datname for row in self.con.execute(
-                'SELECT datname FROM pg_database WHERE NOT datistemplate'
-            )
-        ]
+        """list databases here means list schemas"""
+        return self.inspector.get_schema_names()
```

Comment:
I wouldn't worry about the fact that most other databases supported in ibis do not have the notion of the schema or they call it something else. If it turns out that users move from impala to postgres often, then we can alias `ImpalaClient.list_schemas()` to `list_databases()`.